experimental/c-cpp/post-process.py

```diff
@@ -0,0 +1,213 @@
+  with open(log_path, 'r', encoding='ISO-8859-1') as libfuzzer_out:
+    cov_history = []
+    addr_sanitizer_lines = []
+    for line in libfuzzer_out:
+      edge_coverage = _get_edge_cov_from_line(line)
+      if edge_coverage is not None:
+        cov_history.append(edge_coverage)
+      if 'ERROR: AddressSanitizer' in line:
+        addr_sanitizer_lines.append(line)
+
+  return {'cov_history': cov_history, 'addr_san': addr_sanitizer_lines}
+
+
+def oss_fuzz_out_dir(oss_fuzz_dir: str) -> str:
+  """Returns the out folder of a given OSS-Fuzz repository."""
+  return os.path.join(oss_fuzz_dir, 'build', 'out')
+
+
+def get_oss_fuzz_generated_projects(oss_fuzz_dir: str) -> List[str]:
+  """Gets the sample sessions for auto-gen projects in a given OSS-Fuzz dir."""
+  project_indices = []
+  for project_out in os.listdir(oss_fuzz_out_dir(oss_fuzz_dir)):
+    if project_out.startswith(constants.PROJECT_BASE):
+      try:
+        project_indices.append(
+            int(project_out.replace(constants.PROJECT_BASE, '')))
+      except ValueError:
+        pass
+
+  sorted_project_folders = [
+      constants.PROJECT_BASE + '%d' % (idx)
+      for idx in list(sorted(project_indices))
+  ]
+  return sorted_project_folders
+
+
+def interpret_single_harness_setup(
+    oss_fuzz_dir: str, project_folder_out: str, generated_sample_setup: str,
+    generated_sample_setup_dir: str) -> Optional[Dict[str, Any]]:
+  """Analyses a single auto-generated set up within an auto-generated
+  session."""
+  fuzz_run = os.path.join(generated_sample_setup_dir, 'fuzz-run.err')
+  if not os.path.isfile(fuzz_run):
+    return None
+
+  run_results = interpret_llvm_run(fuzz_run)
+  run_results['target'] = fuzz_run
+
+  # Check the covreage run
+  coverage_proj = os.path.basename(
+      project_folder_out) + "-cov-" + generated_sample_setup
+  harness_run_stats = os.path.join(oss_fuzz_out_dir(oss_fuzz_dir),
+                                   coverage_proj, 'fuzzer_stats', 'fuzzer.json')
+  if os.path.isfile(harness_run_stats):
+    with open(harness_run_stats, 'r') as stats_file:
+      cov_stats = json.load(stats_file)
+    lines_coverage = cov_stats['data'][0]['totals']['lines']['percent']
+  else:
+    lines_coverage = -1.0
+  run_results['line_coverage'] = lines_coverage
+
+  # if there are no results then return None
+  if len(run_results['cov_history']) <= 1:
+    return None
+  return run_results
+
+
+def extract_results_from_generated_samples(
+    oss_fuzz_dir: str, project_folder_out: str) -> List[Dict[str, Any]]:
+  """Iterates an auto-gen session and returns a list of successful results."""
+  total_outcomes = []
+  for generated_sample_setup in os.listdir(
+      os.path.join(project_folder_out, constants.SHARED_MEMORY_RESULTS_DIR)):
+    generated_sample_setup_dir = os.path.join(
+        project_folder_out, constants.SHARED_MEMORY_RESULTS_DIR,
+        generated_sample_setup)
+    #print("Analysing %s"%(dd))
```

Comment:
Remove debug print, this was logged at another place.

Reply:
done